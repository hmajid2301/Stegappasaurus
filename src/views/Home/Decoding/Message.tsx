import * as React from "react";
import { NavigationScreenProp } from "react-navigation";

import ImageMessage from "~/components/ImageMessage";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
}

interface IState {
  isDecoded: boolean;
  message: string;
  photo: string;
}

class Progress extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");
    const message = navigation.getParam("message", "NO-ID");

    this.state = {
      isDecoded: false,
      message,
      photo: uri
    };
  }

  public render() {
    return (
      <ImageMessage
        editable={false}
        message={this.state.message}
        photo={this.state.photo}
      />
    );
  }
}

export default Progress;
