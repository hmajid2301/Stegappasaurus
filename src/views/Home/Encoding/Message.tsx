import { Toast } from "native-base";
import React, { Component } from "react";
import { Image } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import ImageMessage from "~/components/ImageMessage";
import Steganography from "~/services/steganography";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
}

interface IState {
  photo: string;
}

export default class Message extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");

    this.state = {
      photo: uri
    };
  }

  public render() {
    return <ImageMessage action={this.onSubmit} photo={this.state.photo} />;
  }

  private onSubmit = (message: string) => {
    let bitsCanEncode = 0;
    Image.getSize(
      this.state.photo,
      (width, height) => {
        bitsCanEncode = (width * height * 3) / 4;
      },
      () => null
    );

    const binaryMessage = new Steganography(
      "LSB-PNG",
      []
    ).convertMessageToBinary(message);
    const messageLength = binaryMessage.join("").length;

    if (messageLength > bitsCanEncode) {
      Toast.show({
        buttonText: "Okay",
        duration: 5000,
        text: "Message too large to encode in image.",
        type: "danger"
      });
    } else {
      this.props.navigation.navigate("Progress", {
        message,
        uri: this.state.photo
      });
    }
  };
}
