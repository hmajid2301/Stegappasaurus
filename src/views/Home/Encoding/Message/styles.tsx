import { Dimensions, StyleSheet } from "react-native";

import { colors, fonts } from "~/constants";

const SCREEN_HEIGHT = Dimensions.get("screen").height;
const SCREEN_WIDTH = Dimensions.get("screen").width;

const styles = StyleSheet.create({
  container: {
    alignItems: "stretch",
    backgroundColor: "white",
    height: SCREEN_HEIGHT * 0.55,
    justifyContent: "center",
    marginHorizontal: SCREEN_WIDTH * 0.025,
    marginVertical: SCREEN_HEIGHT * 0.25,
    paddingHorizontal: 20,
    paddingVertical: 20
  },

  modalHeader: {
    fontFamily: fonts.header,
    fontSize: 25,
    marginBottom: 20,
    textAlign: "center"
  },

  sliderContainer: {
    alignItems: "stretch",
    flex: 1,
    justifyContent: "center",
    marginBottom: 20
  },

  modalText: {
    fontFamily: fonts.bodyLight
  },

  passwordContainer: {
    flex: 1
  },

  passwordTextInput: {
    borderBottomColor: "grey",
    borderBottomWidth: 1
  },

  buttonContainer: {
    flex: 1,
    marginTop: 20
  }
});

export default styles;
