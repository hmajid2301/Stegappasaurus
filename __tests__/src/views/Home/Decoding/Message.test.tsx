import { shallow } from "enzyme";
import React from "react";

import Message from "~/views/Home/Decoding/Message";

describe("Decoding Message: Match Snapshots", () => {
  test("1", () => {
    const navigate = {
      navigate: jest.fn(),
      getParam: jest.fn()
    };
    const component = shallow(<Message navigation={navigate as any} />);
    expect(component).toMatchSnapshot();
  });
});
