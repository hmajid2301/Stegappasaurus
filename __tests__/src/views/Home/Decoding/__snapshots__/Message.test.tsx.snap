// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decoding Message: Match Snapshots 1 1`] = `
<ImageMessage
  action={null}
  editable={false}
  message=""
  navigation={
    Object {
      "getParam": [MockFunction] {
        "calls": Array [
          Array [
            "uri",
            "NO-ID",
          ],
          Array [
            "message",
            "NO-ID",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "navigate": [MockFunction],
    }
  }
/>
`;
